src/hnsw/build.c

```diff
@@ -482,25 +495,51 @@ static void BuildIndex(
         assert(error == NULL);
     }
 
-    char *result_buf = NULL;
-    usearch_save(buildstate->usearch_index, NULL, &result_buf, &error);
-    assert(error == NULL && result_buf != NULL);
-
-    size_t num_added_vectors = usearch_size(buildstate->usearch_index, &error);
+    metadata = usearch_metadata(buildstate->usearch_index, &error);
     assert(error == NULL);
 
+    if(buildstate->index_file_path == NULL) {
+        // Save index into temporary file
+        // To later mmap it into memory
+        // Filename is /tmp/ldb-index-$relfilenode.bin
+        // The file will be removed in the end
+        tmp_index_file_path = palloc0(tmp_index_file_char_cnt);
+        snprintf(tmp_index_file_path, tmp_index_file_char_cnt, tmp_index_file_fmt_str, index->rd_rel->relfilenode);
+        usearch_save(buildstate->usearch_index, tmp_index_file_path, NULL, &error);
+        assert(error == NULL);
+        index_file_fd = open(tmp_index_file_path, O_RDONLY);
+    } else {
+        index_file_fd = open(buildstate->index_file_path, O_RDONLY);
+    }
+    assert(index_file_fd > 0);
+
+    num_added_vectors = usearch_size(buildstate->usearch_index, &error);
+    assert(error == NULL);
     elog(INFO, "done saving %ld vectors", num_added_vectors);
 
+    //****************************** mmap index to memory BEGIN ******************************//
+    usearch_free(buildstate->usearch_index, &error);
+    assert(error == NULL);
+    buildstate->usearch_index = NULL;
+
+    fstat(index_file_fd, &index_file_stat);
+    result_buf = mmap(NULL, index_file_stat.st_size, PROT_READ, MAP_PRIVATE, index_file_fd, 0);
+    assert(result_buf != MAP_FAILED);
+    //****************************** mmap index to memory END ******************************//
+
     //****************************** saving to WAL BEGIN ******************************//
     UpdateProgress(PROGRESS_CREATEIDX_PHASE, PROGRESS_HNSW_PHASE_LOAD);
-    StoreExternalIndex(index, buildstate->usearch_index, forkNum, result_buf, &opts, num_added_vectors);
-
+    StoreExternalIndex(index, &metadata, forkNum, result_buf, &opts, num_added_vectors);
     //****************************** saving to WAL END ******************************//
 
-    usearch_free(buildstate->usearch_index, &error);
-    free(result_buf);
-    assert(error == NULL);
-    buildstate->usearch_index = NULL;
+    assert(munmap(result_buf, index_file_stat.st_size) == 0);
```

Comment:
in `release` builds `assert`s are compiled away (it is a macro, not a function, so everything in the parentheses gets dropped).
So, the munmap call has to be outside of `assert`